lib/hanami/assets/helpers.rb

```diff
@@ -186,9 +240,25 @@ def javascript(*sources)
       #   <%= stylesheet 'application' %>
       #
       #   # <link href="https://assets.bookshelf.org/assets/application-28a6b886de2372ee3922fcaf3f78f2d8.css" type="text/css" rel="stylesheet">
-      def stylesheet(*sources)
+      def stylesheet(*sources, **options)
         _safe_tags(*sources) do |source|
-          html.link(href: _typed_asset_path(source, STYLESHEET_EXT), type: STYLESHEET_MIME_TYPE, rel: STYLESHEET_REL).to_s
+          tag_options = options.merge({
+            href: _typed_asset_path(source, STYLESHEET_EXT),
+            type: STYLESHEET_MIME_TYPE,
+            rel: STYLESHEET_REL
+          })
+
+          if _sri? || tag_options.include?(:integrity)
+            tag_options[:integrity] ||= _sri_value(source, STYLESHEET_EXT)
+            tag_options[:crossorigin] ||= CROSSORIGIN_ANONYMOUS
+          end
+
+          html.link(
+            href: tag_options.delete(:href),
```

Comment:
@accuser I know it's annoying to change existing tests, but this creates an overhead of instantiate a Hash and we can avoid it.

Do you think that giving a specific order to the Hash saves us from changing the tests? If not, please change the tests. It's a boring task, but it will pay in terms of performance. Thanks for understanding. :smile: